crypto/keyring/keyring_test.go

```diff
@@ -1153,6 +1153,52 @@ func TestBackendConfigConstructors(t *testing.T) {
 	require.Equal(t, "keyring-test", backend.PassPrefix)
 }
 
+func TestRenameKey(t *testing.T) {
```

Comment:
For each test, should we check more than just an error? Should we also check whether each key does or does not exist within the keyring after renaming?

Reply:
yeah, i think i'll refactor this into a table test and flesh out the checks some more

Reply:
>  Should we also check whether each key does or does not exist within the keyring after renaming?

Reply:
Within this test, rename is called three times at the end without checking the specific error thrown or what keys exist after each failed attempt. I was thinking the extra checks would be added to the failed attempts. At the least, I think we should check to make sure we are receiving the expected error and not just any error.

Reply:
[refactored](https://github.com/cosmos/cosmos-sdk/pull/9601/files#diff-87e627ad16ac7fbc8eab2e3c33468a4c734661c5e6e4a603b8cb35c3ac2b8f07R1156-R1212) the tests. should be a bit more concise. let me know if its missing anything please!